### ISSUE TITLE
Color to normals

### ISSUE THREAD

**Issue by Jcastanyo** [vocdex/digit-depth#3]
Hi! I am Julio again, I create this issue to ask you about the normals. How do you get the normals from the images for the ground truth?
Thanks in advance!

**Comment by vocdex** [comment]
To obtain normals, we use a small sphere (D=6mm) and press it on DIGIT 40-50 times. We use sphere because it is possible to get ground truth normals for the sphere due to its geometry. Please refer to this [script](https://github.com/vocdex/digit-depth/blob/main/digit_depth/dataio/generate_sphere_gt_normals.py) for the implementation. To understand logic behind it, please check section 3.5 of GelSight [Paper](https://www.mdpi.com/1424-8220/17/12/2762#:~:text=The%20sensor%20calibration,%2C1)
In case you will be using this repo, you can label sphere radius and center based on /scripts/label_data.py and after this, you can simply execute create_image_dataset.py to obtain folders /datasets/A, datasets/B with ground truth normal images for spheres. 
Hope it was helpful.

**Comment by Jcastanyo** [comment]
Thanks!! I have one more question. If you train a neural network to calculate the normals, but the training set only contains normals from the sphere, It will generalize to other types of geometries?
Thank you very much for your help!

**Comment by vocdex** [comment]
Yes, it will generalize quite well because we are only using sphere to learn a general relationship between RGB intensity and normals. Please refer to [this](https://arxiv.org/pdf/2111.07524.pdf) and [this](http://gelsight.csail.mit.edu/wedge/ICRA2021_Wedge.pdf) papers' relevant sections for more info on generalization.

**Comment by Jcastanyo** [comment]
Ok, I will take a look at these papers. Thank you again for your attention and help!
